core/transcoder.go

```diff
@@ -41,6 +42,36 @@ func NewUnrecoverableError(err error) UnrecoverableError {
 
 var WorkDir string
 
+func setEffectiveDetectorConfig(md *SegTranscodingMetadata) {
+	aiEnabled := DetectorProfile != nil
+	actualProfile := DetectorProfile
+	if aiEnabled {
+		presetSampleRate := DetectorProfile.(*ffmpeg.SceneClassificationProfile).SampleRate
+		if md.DetectorEnabled && len(md.DetectorProfiles) == 1 {
+			actualProfile = md.DetectorProfiles[0]
+			requestedSampleRate := actualProfile.(*ffmpeg.SceneClassificationProfile).SampleRate
+			// 0 is not a valid value
+			if requestedSampleRate == 0 {
+				requestedSampleRate = math.MaxUint32
+			}
+			actualProfile.(*ffmpeg.SceneClassificationProfile).SampleRate = uint(math.Min(float64(presetSampleRate),
+				float64(requestedSampleRate)))
+			// copy other fields from default AI capability, as we don't yet support custom ones
+			actualProfile.(*ffmpeg.SceneClassificationProfile).ModelPath = DetectorProfile.(*ffmpeg.SceneClassificationProfile).ModelPath
+			actualProfile.(*ffmpeg.SceneClassificationProfile).Input = DetectorProfile.(*ffmpeg.SceneClassificationProfile).Input
+			actualProfile.(*ffmpeg.SceneClassificationProfile).Output = DetectorProfile.(*ffmpeg.SceneClassificationProfile).Output
+			actualProfile.(*ffmpeg.SceneClassificationProfile).Classes = DetectorProfile.(*ffmpeg.SceneClassificationProfile).Classes
+		}
+	}
+	if actualProfile != nil && actualProfile.(*ffmpeg.SceneClassificationProfile).SampleRate < math.MaxUint32 {
```

Comment:
What's the reason for comparison to `MaxUint32` rather than having it be zero?

Reply:
That value ends up in something like `select=not(mod(Nframe, SampleRate))` in Ffmpeg filter, so we don't want to have 0 here. Hence, MAX as default value.

Reply:
👍  Gotcha